hnn_core/network_builder.py

```diff
@@ -425,35 +437,184 @@ def _create_cells_and_feeds(self):
             else:
                 _PC.cell(gid, feed_cell.nrn_netcon)
 
+    def _connect_celltypes(self, src_type, target_type, loc,
+                           receptor, nc_dict, unique=False,
+                           allow_autapses=True):
+        """Connect two cell types for a particular receptor.
+
+        Parameters
+        ----------
+        src_type : str
+            The source cell type
+        target_type : str
+            The target cell type
+        loc : str
+            If 'proximal' or 'distal', the corresponding
+            dendritic sections from Cell.sect_loc['proximal']
+            or Cell.Sect_loc['distal'] are used
+        receptor : str
+            The receptor.
+        nc_dict : dict
+            The connection dictionary containing keys
+            A_delay, A_weight, lamtha, and threshold.
+        unique : bool
+            Is unique?
+        allow_autapses : bool
+            If True, allow connecting neuron to itself.
+        """
+        net = self.net
+        connection_name = f'{src_type}_{target_type}_{receptor}'
+        if connection_name not in self.ncs:
+            self.ncs[connection_name] = list()
+        for gid_target in net.gid_dict[_long_name(target_type)]:
+            if _PC.gid_exists(gid_target):
+                gid_srcs = net.gid_dict[_long_name(src_type)]
+                if unique:
+                    gid_srcs = [gid_target + net.gid_dict[src_type][0]]
```

Comment:
This feels a bit convoluted. Is this here to assign the subset of source gids that apply to the given target cell type?

Reply:
see [comment here](https://github.com/jonescompneurolab/hnn-core/pull/164/files#r487466791). The source gids from net.gid_dict are actually ignored if `unique=True`